br/pkg/backup/client.go

```diff
@@ -1168,8 +1183,20 @@ func (bc *Client) fineGrainedBackup(
 		for {
 			select {
 			case err := <-errCh:
-				// TODO: should we handle err here?
-				return errors.Trace(err)
+				if !berrors.Is(err, berrors.ErrFailedToConnect) {
+					return errors.Trace(err)
+				}
+				storeErr, ok := err.(*StoreBasedErr)
+				if !ok {
+					return errors.Trace(err)
+				}
+
+				storeID := storeErr.storeID
+				message := storeErr.message
+				maxDisconnect[storeID]++
+				if maxDisconnect[storeID] > 3 {
```

Comment:
nit, maybe make it a const?

Reply:
A better way is reusing the `backupRetryTimes`
Done